lfs.c

```diff
@@ -4246,11 +4415,26 @@ static int lfs_rawformat(lfs_t *lfs, const struct lfs_config *cfg) {
             return err;
         }
 
+        if(cfg->block_count == 0){
+            // Attempt to read a (possibly) prior superblock
```

Comment:
formatting is also a common operation if the user just wants to wipe all files in the filesystem. In that case, providing 0 is a convenient way of doing that.

Reply:
Hmm. This can also be accomplished by the following:

``` c
lfs_t lfs;

// mount once to find the block_count
struct lfs_config cfg = {
    .block_size = 0,
    ...
};

err = lfs_mount(&lfs, &cfg);
if (err && err != LFS_ERR_INVAL) {
    return err;
}

if (err != LFS_ERR_INVAL) {
    struct lfs_fsinfo fsinfo;
    err = lfs_fs_stat(&lfs, &fsinfo);
    if (err) {
        return err;
    }
    
    cfg.block_count = fsinfo.block_count;
} else {
    // default must be user provided
    cfg.block_count = DEFAULT_BLOCK_COUNT;
}

// format with either found or default block_count
err = fs_format(&lfs, &cfg);
if (err) {
    return err;
}
```

Effectively moving this code outside of the filesystem.

I'm still not convinced `lfs_format` should be extended in this way. Adding more features to `lfs_format` risks putting users into confusing situations ("try zeroing disk and then formatting") and from my perspective is outside of the scope of `lfs_format`.

Reply:
I think that's a fair assessment; changing it now

Reply:
addressed in [23089d5](https://github.com/littlefs-project/littlefs/pull/866/commits/23089d5758b2a38ffa45028e446de292791272ff)

Reply:
I still think the restructure of `lfs_scan_for_superblocks` and `lfs_scan_for_state_updates` should be reverted.